### ISSUE TITLE
[IMPROVEMENT] Use `swagger_ui_init_oauth` parameter of original function app for versioned /docs (e.g. /v1/docs)

### ISSUE THREAD

**Issue by stoney95** [alexschimpf/fastapi-versionizer#45]
### Subject of the issue
I use `swagger_ui_init_oauth` when creating the `FastAPI` instance. This allows me to use the set values when using the "Authorize" feature of the swagger UI. This is very handy as it allows to set the client id and scopes. 

I use it like this: 

```python
fastapi_app = FastAPI(
    swagger_ui_parameters={"supportedSubmitMethods": allowed_http_methods},
    swagger_ui_init_oauth={
        "clientId": "my-client-id",
        "scopes": "required_scopes",
    },
)

versions = Versionizer(
     app=fastapi_app,
     prefix_format='/v{major}',
     semantic_version_format='{major}',
     latest_prefix='/latest',
     sort_routes=True
).versionize()
```

Applying the `Versionizer` does remove the values in versioned docs. It would be great if they are kept. 
When accessing "example.com/docs" I can see "my-client-id" is set. But when accessing "example.com/v1/docs" I can't see the client id in the Authorize feature.

Screenshots for further details:
- Authorize Feature
<img width="1454" alt="Screenshot 2023-10-23 at 18 48 22" src="https://github.com/alexschimpf/fastapi-versionizer/assets/34100018/df64de7b-2722-448a-8ff7-f96831876eef">

- Client id is set in root docs
<img width="645" alt="Screenshot 2023-10-23 at 18 41 29" src="https://github.com/alexschimpf/fastapi-versionizer/assets/34100018/1f43c46a-8854-4d4c-9557-dadb76aa9819">

- Client id is not set in versioned docs
<img width="646" alt="Screenshot 2023-10-23 at 18 42 35" src="https://github.com/alexschimpf/fastapi-versionizer/assets/34100018/60e8ee75-5ab7-4b47-b37a-c8b227fd5d71">

### Your environment
* 3.9.16
* MacOS 13.2.1

### Steps to reproduce
```
from fastapi import FastAPI
from fastapi_versionizer import Versionizer
from fastapi.openapi.models import OAuthFlowImplicit, OAuthFlows
from fastapi.security import OAuth2

oauth2_scheme = OAuth2(
    flows=OAuthFlows(
        implicit=OAuthFlowImplicit(
            authorizationUrl="https://login.microsoftonline.com/oauth2/v2.0/authorize",
        )
    ),
    auto_error=True,
)

fastapi_app = FastAPI(
    swagger_ui_init_oauth={
        "clientId": "my-client-id",
        "scopes": "required_scopes",
    },
)

versions = Versionizer(
     app=fastapi_app,
     prefix_format='/v{major}',
     semantic_version_format='{major}',
     latest_prefix='/latest',
     sort_routes=True
).versionize()

router = APIRouter(prefix="/test")

def check_token(token: Depends(oauth2_scheme)):
    print(token)

@router.get("", dependencies=[Depends(check_token)])
def print_test():
    print("test")
```

### Expected behaviour
I can use the Authorize feature in the versioned docs.

### Actual behaviour
The values initialized with `swagger_ui_init_oauth` are not available in versioned docs (example.com/v1/docs)

**Comment by alexschimpf** [comment]
Sure, I'll whip up a fix for this soon and let you know. Thanks for finding this!

**Comment by alexschimpf** [comment]
Alright, this should now be fixed in v3.0.2. Please let me know if you find any other problems.

**Comment by stoney95** [comment]
@alexschimpf, Thanks for taking care of that issue so quickly! 

Now I am able to see the client-id in the Authorize feature. I also tested it with our setup. I get forwarded to the right login page and can login in there -> scopes are also set correctly. 

But this runs into another issue. The redirect url cannot be found. This happens with the root docs (/docs) as well as with the versioned docs (/v1/docs). It appears as the `Versionizer` overwrites / adjusts the `swagger_ui_oauth2_redirect_url` which defaults to "/docs/oauth2-redirect". 

When trying to login via the "Authorize" button I see the following
<img width="847" alt="Screenshot 2023-10-25 at 09 16 01" src="https://github.com/alexschimpf/fastapi-versionizer/assets/34100018/77d60960-387c-47cf-8754-8d79180dc149">

This refers to a login I did via the root docs

**Comment by alexschimpf** [comment]
Ok I'll try to look into this once I get a chance

**Comment by alexschimpf** [comment]
Ah okay. I think I see the issue. For the main docs page, it happens because the redirect route is being stripped out in the `_strip_routes` function. Then it's not getting added to the versioned docs because the redirect route is not of type `APIRoute` and is thus getting ignored.

Let me try to get a fix out for this.

**Comment by alexschimpf** [comment]
Sorry for the delay. I need to get back to this. I had a solution mostly written some time ago, but never got around to pushing it. Hopefully I can get something pushed this week.

**Comment by alexschimpf** [comment]
@stoney95 I think this PR should do this trick: https://github.com/alexschimpf/fastapi-versionizer/pull/50
Could you check this to see if the behavior is expected?

**Comment by stoney95** [comment]
@alexschimpf, a happy new year to you! 

Thanks for providing the fix. I ran some local tests and the issues from are above are resolved 🚀

**Comment by alexschimpf** [comment]
Sorry for the long (ok, very long) delay. Been a busy bee. This fix is available in v4.0.1.
